### PR TITLE
Start of testing infrastructure

### PR DESCRIPTION
**Pull request by bookdude13**
Starting to write tests. My general approach is to take advantage of the DAO paradigm that is set up and create testing implementations of the DAOs. These test DAOs will simply call their supplied functions instead of the trait/interface's functions. 

Also thinking that, for the integration tests in the tests/ directory, each publicly available command will have its own file containing all tests related to that command. Open for discussion on alternate ways to structure this.

### CODE REVIEW

File: tests/dao/project_dao_testing.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+			new_project_fn: Box::new(|_, _| -> Result<Project, Error> { Err(Error::TodoErr) }),
+			get_project_fn: Box::new(|_| -> Result<Project, Error> { Err(Error::TodoErr) }),
+			update_project_fn: Box::new(|_| -> Result<(), Error> { Err(Error::TodoErr) })
+		}
+	}
+}
+
+/// The Dao implementation simply calls the corresponding stored function
+impl ProjectDao for ProjectDaoTesting {
+	fn new_project(&self, name: &str, layoutid: u32) -> Result<Project, Error> {
```
Comment (computergeek125) [inline tests/dao/project_dao_testing.rs:32]: I know it's picky, but the indentation isn't consistent here and a couple of other places.  Does that matter much too much to you?

File: tests/dao/permission_dao_testing.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			add_initial_permission_fn: Box::new(|_| -> Result<(), Error> { Err(Error::TodoErr) }),
+			get_all_permissions_fn: Box::new(|_| -> Result<Vec<Permission>, Error>  { Err(Error::TodoErr) }),
+			get_permission_fn: Box::new(|_| -> Result<Permission, Error> { Err(Error::TodoErr) }),
+		}
+	}
+}
+
+/// The Dao implementation simply calls the corresponding stored function
+impl PermissionDao for PermissionDaoTesting {
+	fn add_initial_permission(&self, root_uid: u32) -> Result<(), Error> {
```
Comment (computergeek125) [inline tests/dao/permission_dao_testing.rs:32]: See other comment about indentation

File: tests/dao/layout_dao_testing.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+			get_last_layout_fn: Box::new(|_| -> Result<Layout, Error> { Err(Error::TodoErr) }),
+			layout_exists_fn: Box::new(|_| -> Result<bool, Error> { Err(Error::TodoErr) }),
+			patch_channel_fn: Box::new(|_, _, _| -> Result<u64, Error> { Err(Error::TodoErr) })
+		}
+	}
+}
+
+/// The Dao implementation simply calls the corresponding stored function
+impl LayoutDao for LayoutDaoTesting {
+	fn new_layout(&self, name: &str, fixtures: Vec<u32>) -> Result<Layout, Error> {
```
Comment (computergeek125) [inline tests/dao/layout_dao_testing.rs:38]: See other comment about indentation (double function here)

File: tests/dao/data_dao_testing.rs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+			new_data_fn: Box::new(|_, _, _| -> Result<(), Error>  { Err(Error::TodoErr) }),
+			get_data_fn: Box::new(|_, _| -> Result<Vec<u16>, Error> { Err(Error::TodoErr) }),
+			update_data_fn: Box::new(|_, _, _| -> Result<(), Error> { Err(Error::TodoErr) }),
+		}
+	}
+}
+
+/// The Dao implementation simply calls the corresponding stored function
+impl DataDao for DataDaoTesting {
+	fn new_data_default(
```
Comment (computergeek125) [inline tests/dao/data_dao_testing.rs:33]: See other comment about indentation.  Just the first line here, I think

File: tests/dao/sequence_dao_testing.rs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+			set_layout_fn: Box::new(|_, _| -> Result<(), Error> { Err(Error::TodoErr) }),
+			sequence_exists_fn: Box::new(|_| -> Result<bool, Error> { Err(Error::TodoErr) }),
+			get_channel_ids_fn: Box::new(|_| -> Result<Vec<u32>, Error> { Err(Error::TodoErr) })
+		}
+	}
+}
+
+/// The Dao implementation simply calls the corresponding stored function
+impl SequenceDao for SequenceDaoTesting {
+	fn get_sequence(&self, seqid: u32) -> Result<Sequence, Error> {
```
Comment (computergeek125) [inline tests/dao/sequence_dao_testing.rs:38]: See other comment about indentation

**Review by computergeek125 — APPROVED** [review]
Looks good other than the interesting indents

File: tests/dao/project_dao_testing.rs (hunk 1 above)
Comment (bookdude13) [reply tests/dao/project_dao_testing.rs:32]: Not sure, but I think that's just GitHub's view of it. Doesn't look strange to me in Sublime...

**Pull request merged**